includes/classes/SiteAutomation/Integration.php

```diff
@@ -52,28 +68,43 @@ public function get_curated_posts( $posts, $query ) {
 		}
 
 		$curated_response = $this->request->get( $query_vars['sophi_curated_page'], $query_vars['sophi_curated_widget'] );
-
-		// Determine how we should format the results  based on the fields parameter.
-		$fields = $query->get( 'fields', '' );
-
-		switch ( $fields ) {
-			case 'ids':
-				$new_posts = $this->format_hits_as_ids( $curated_response );
-				break;
-
-			case 'id=>parent':
-				$new_posts = $this->format_hits_as_id_parents( $curated_response );
-				break;
-
-			default:
-				$new_posts = $this->format_hits_as_posts( $curated_response );
-				break;
-		}
-
-		$new_posts = array_filter( $new_posts );
-
-		if ( ! empty( $new_posts ) ) {
-			return $new_posts;
+		$request_status   = $this->request->get_status();
+
+		if ( $request_status['success'] ) {
+			$query->sophi_curated_post_list_success = true;
+			$query->num_posts = count( $curated_response );
+
+			// Determine how we should format the results  based on the fields parameter.
+			$fields = $query->get( 'fields', '' );
+
+			switch ( $fields ) {
+				case 'ids':
+					$new_posts = $this->format_hits_as_ids( $curated_response );
+					break;
+
+				case 'id=>parent':
+					$new_posts = $this->format_hits_as_id_parents( $curated_response );
+					break;
+
+				default:
+					$new_posts = $this->format_hits_as_posts( $curated_response );
+					break;
+			}
+
+			/**
+			 * The curated post list result that is injected to WP_Query.
+			 *
+			 * @since 1.0.9
+			 * @hook sophi_curated_post_list
+			 *
+			 * @param {array} $new_posts Post list.
+			 * @param {string} $query_vars['sophi_curated_page'] Sophi curated page param.
+			 * @param {string} $query_vars['sophi_curated_widget'] Sophi curated widget param.
+			 * @param {object} $query Original query.
+			 *
+			 * @return {integer} Posts per page limit.
+			 */
+			return apply_filters( 'sophi_curated_post_list', array_filter( $new_posts ), $query_vars['sophi_curated_page'], $query_vars['sophi_curated_widget'], $query );
```

Comment:
@oscarssanchez This filter should probably be added to the final `return $posts` on line 110. It now will only run if `$request_status['success']` is true, but and end user might still want to (or even need this filter even more) if `$request_status['success']` is false.

Reply:
This makes sense to me @barryceelen I have adjusted in this commit: https://github.com/globeandmail/sophi-for-wordpress/pull/183/commits/8aa65f6c04230d7e4a1f9822eccecfe266c208fd and added also the `$request_status`, it could be handy to have if the request failed to check the errors. Let me know if it makes sense to you.

Reply:
Looks good to me, thanks!